otel-extension/src/main/java/io/scalac/mesmer/otelextension/MesmerAkkaStreamInstrumentationModule.java

```diff
@@ -0,0 +1,90 @@
+package io.scalac.mesmer.otelextension;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import io.opentelemetry.javaagent.tooling.muzzle.InstrumentationModuleMuzzle;
+import io.opentelemetry.javaagent.tooling.muzzle.VirtualFieldMappingsBuilder;
+import io.opentelemetry.javaagent.tooling.muzzle.references.ClassRef;
+import io.scalac.mesmer.agentcopy.akka.stream.AkkaStreamAgent;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import net.bytebuddy.matcher.ElementMatchers;
+
+import java.util.*;
+
+import static net.bytebuddy.matcher.ElementMatchers.isConstructor;
+
+@AutoService(InstrumentationModule.class)
+public class MesmerAkkaStreamInstrumentationModule extends InstrumentationModule implements InstrumentationModuleMuzzle {
```

Comment:
Hot stuff - nice!

From the InstrumentationModuleMuzzle javadoc: 

```
/**
 * This interface contains methods that muzzle automatically adds to the {@link
 * InstrumentationModule}. They are not supposed to be used by end-users, only by our own internal
 * code.
 */
```

I suspect that this will still work with otel extension but should we worry? If for some reason this will stop being supported, will our extension still have ways to work properly? We can add the jar with helper classes to the classpath manually, right?

Reply:
`getMuzzleHelperClassNames` is part for the `InstrumentationModule` api, so we're safe. The only functionality that we use from `InstrumentationModuleMuzzle` is :
```
@Override
    public void registerMuzzleVirtualFields(VirtualFieldMappingsBuilder builder) {
        builder.register("akka.stream.impl.fusing.GraphInterpreter$Connection", "scala.Tuple2");
    }
```
And it's to make opentelemetry actualy turn virtual field into actual field on the class